### ISSUE TITLE
Feature request: Signed Distance Field for fonts and monocolor shapes

### ISSUE THREAD

**Issue by akk0rd87** [libsdl-org/SDL#6419]
I would like to know if it is possible to add this feature to SDL for support different drawing backends: opengl(es), Metal, Vulkan and etc.? It is cool technique for drawing scalable fonts and monocolor shapes without texture rasterization. All scaling is handled by GPU, not CPU.

See more:
https://libgdx.com/wiki/graphics/2d/fonts/distance-field-fonts
https://steamcdn-a.akamaihd.net/apps/valve/2007/SIGGRAPH2007_AlphaTestedMagnification.pdf

**Comment by slouken** [comment]
This is already supported in SDL_ttf, checkout `TTF_SetFontSDF()`

**Comment by akk0rd87** [comment]
I asked about SDF [here](https://discourse.libsdl.org/t/sdl2-ttf-2-0-18-prerelease/34240/11) and [Sylvain_Becker](https://discourse.libsdl.org/u/Sylvain_Becker) pointed me to custom example for opengles only. And those code and shaders seems had written outside SDL, just as an example.
It means it will be pain if someone wants to combine SDL_Renderer and code like this written by himself. Every change in src/render/ source may broke this compability.

I'm using my SDF shaders over opengles only and compability with SDL_Renderer was broken at least two times:
- when glGenBuffers was added to src/render/opengles2/SDL_render_gles2.c
- when glGenBuffers was removed from src/render/opengles2/SDL_render_gles2.c

Second problem is that opengles is deprecated on ios, and I need to move SDF to metal. And I'm not sure if I can do it by myself.

**Comment by 1bsyl** [comment]
yes, that would be possible if user can provide its own shaders to SDL. Currently shaders are specific to back-end, provided as source or binaries, so there are more platform code
Hopefully with Ryan's universal shader language, that would make things easier.

**Comment by slouken** [comment]
We've added custom shader support for the GPU renderer, so you can easily do this. We've added an MSDF font example here, so you can see it in action: https://github.com/libsdl-org/SDL/commit/2aee105b43f77ef0a84e7d9c5acab29a9ca0d7b5
